spring-integration-core/src/main/java/org/springframework/integration/endpoint/AbstractPollingEndpoint.java

```diff
@@ -266,7 +267,17 @@ private boolean doPoll() {
 			if (holder != null) {
 				holder.setMessage(message);
 			}
-			this.handleMessage(message);
+			try {
+				this.handleMessage(message);
+			}
+			catch (Exception e) {
+				if (e instanceof MessagingException) {
+					throw new OriginalMessageContainingMessagingException(message, (MessagingException) e);
+				}
+				else {
+					throw new MessagingException(message, e);
+				}
```

Comment:
The intention is to enhance the error message so that it has the `failedMessage` (the message at the point of failure) and the `originalMessage` (the message that started the flow) - it could have undergone many transformations before it failed.